src/src/PluginStructs/P025_data_struct.cpp

```diff
@@ -16,56 +20,41 @@ int16_t P025_data_struct::read() {
   config |= static_cast<uint16_t>(mux) << 12;
   config |= (0x8000); // Start a single conversion
 
-  Wire.beginTransmission(i2cAddress);
-  Wire.write((uint8_t)(0x01));   // access to config register
-  Wire.write((uint8_t)(config >> 8));
-  Wire.write((uint8_t)(config & 0xFF));
-  Wire.endTransmission();
+  I2C_write16_reg(i2cAddress, P025_CONFIG_REGISTER, config);
+
+
+  // See https://github.com/letscontrolit/ESPEasy/issues/3159#issuecomment-660546091
+  const unsigned long timeout = millis() + 10;
 
-  while (isReady025()==false) delay(1);
-//  delay(9); // See https://github.com/letscontrolit/ESPEasy/issues/3159#issuecomment-660546091
-  return readConversionRegister025();
+  while (!timeOutReached(timeout) && !isReady025()) {
+    delay(1);
+  }
+
+  // Conversion register represents 2-complement format.
+  return (int16_t)readConversionRegister025();
 }
 
-uint8_t P025_data_struct::getMux()
+uint8_t P025_data_struct::getMux() const
 {
```

Comment:
In an earlier comment I suggested to remove this access method, as it is only used once in a log statement, where the actual setting could have also been used. It would scrape off a few bytes from the bin size 😉